pgxn/neon/pagestore_smgr.c

```diff
@@ -3698,6 +3698,9 @@ neon_write(SMgrRelation reln, ForkNumber forknum, BlockNumber blocknum, const vo
 #ifndef DEBUG_COMPARE_LOCAL
 			/* This is a bit tricky. Check if the relation exists locally */
 			if (mdexists(reln, forknum))
+#else
+			if (mdexists(reln, INIT_FORKNUM))
+#endif
 			{
 				/* It exists locally. Guess it's unlogged then. */
```

Comment:
Hmm, could we check INIT_FORKUM also in !DEBUG_COMPARE_LOCAL mode? 

Does this handle indexes that are going through an "unlogged build" correctly? They don't have an init fork.

Reply:
> Hmm, could we check INIT_FORKUM also in !DEBUG_COMPARE_LOCAL mode?

I thought about it - actually init fork should exists for unlogged relations.
But I preferred to leave everything as it is (working - don't touch:) and changed only DEBUG_COMPARE_LOCAL mode.

> Does this handle indexes that are going through an "unlogged build" correctly? They don't have an init fork.
I explicitly create init fork in `neon_start_unlogged_build` and remove in `neon_end_unlogged_build`